schema/encoding/jsonschema/benchmark_complex_test.go

```diff
@@ -0,0 +1,221 @@
+package jsonschema_test
+
+import (
+	"bytes"
+	"github.com/rs/rest-layer/schema"
+	"github.com/rs/rest-layer/schema/encoding/jsonschema"
+	"math"
+	"testing"
+	"time"
+)
+
+// Optional schema field
+func Optional(s schema.Field) schema.Field {
+	s.Required = false
+	return s
+}
+
+// Default modifies an existing schema.Field with a default value.
+func Default(f schema.Field, v interface{}) schema.Field {
+	f.Default = v
+	return f
+}
+
+// Required modifies an existing schema.Field to be required.
+func Required(s schema.Field) schema.Field {
+	s.Required = true
+	return s
+}
+
+func rfc3339Nano() schema.Field {
```

Comment:
This could "safely" be a var, just like [schema.IDField](https://godoc.org/github.com/rs/rest-layer/schema#pkg-variables)

Since it's not a pointer value, it will be copied on assignment, so that you could still do:

```go
...
s := Schema{
        Fields: schema.Fields{
                "x": Required(rfc3339Nano),
        },
}
```